mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala

```diff
@@ -879,6 +879,7 @@ object Matrices {
       case dm: BDM[Double] =>
         new DenseMatrix(dm.rows, dm.cols, dm.data, dm.isTranspose)
       case sm: BSM[Double] =>
+        sm.compact()
```

Comment:
I'm not sure if this would be regarded as an error for Breeze, since they did provide a compact function. I'll try to check with Breeze.

Close the PR for now.